src/test/java/io/strimzi/kafka/bridge/http/services/ConsumerService.java

```diff
@@ -0,0 +1,193 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ConsumerService extends BaseService {
+
+    public ConsumerService(WebClient webClient) {
+        super(webClient);
+    }
+
+    // Consumer basic requests
+
+    public HttpRequest<JsonObject> createConsumerRequest(String groupId, JsonObject json) {
+        return postRequest(Urls.consumer(groupId))
+                .putHeader(CONTENT_LENGTH, String.valueOf(json.toBuffer().length()))
+                .putHeader(CONTENT_TYPE, BridgeContentType.KAFKA_JSON);
+    }
+
+    HttpRequest<JsonObject> deleteConsumerRequest(String url) {
+        return deleteRequest(url)
+                .putHeader(CONTENT_TYPE, BridgeContentType.KAFKA_JSON)
+                .as(BodyCodec.jsonObject());
+    }
+
+    public HttpRequest<Buffer> consumeRecordsRequest(String groupId, String name) {
+        return getRequest(Urls.consumerInstancesRecords(groupId, name, 1000, null))
+                .putHeader(ACCEPT, BridgeContentType.KAFKA_JSON_JSON);
+    }
+
+    public HttpRequest<JsonObject> subscribeConsumerRequest(String groupId, String name, JsonObject json) {
+        return postRequest(Urls.consumerInstancesSubscription(groupId, name))
+                .putHeader(CONTENT_LENGTH, String.valueOf(json.toBuffer().length()))
+                .putHeader(CONTENT_TYPE, BridgeContentType.KAFKA_JSON);
+    }
+
+    public HttpRequest<JsonObject> offsetsRequest(String groupId, String name, JsonObject json) {
+        return postRequest(Urls.consumerInstancesOffsets(groupId, name))
+                .putHeader(CONTENT_LENGTH, String.valueOf(json.toBuffer().length()))
+                .putHeader(CONTENT_TYPE, BridgeContentType.KAFKA_JSON)
+                .as(BodyCodec.jsonObject());
+    }
+
+    // Consumer actions
+
+    public ConsumerService deleteConsumerSubscription(VertxTestContext context, String groupId, String name, String... topicNames) throws InterruptedException, ExecutionException, TimeoutException {
+        JsonArray topics = new JsonArray();
+        for (String topicName : topicNames) {
+            topics.add(topicName);
+        }
+
+        JsonObject topicsRoot = new JsonObject();
+        topicsRoot.put("topics", topics);
+
+        CompletableFuture<Boolean> unsubscribe = new CompletableFuture<>();
+        deleteRequest(Urls.consumerInstancesSubscription(groupId, name))
+                .putHeader(CONTENT_LENGTH, String.valueOf(topicsRoot.toBuffer().length()))
+                .as(BodyCodec.jsonObject())
+                .sendJsonObject(topicsRoot, ar -> {
+                    context.verify(() -> {
+                        assertTrue(ar.succeeded());
+                        assertEquals(HttpResponseStatus.NO_CONTENT.code(), ar.result().statusCode());
+                        context.completeNow();
+                    });
+                    unsubscribe.complete(true);
+                });
+
+        unsubscribe.get(HTTP_REQUEST_TIMEOUT, TimeUnit.SECONDS);
```

Comment:
I am not pretty sure about this pattern here ... calling the complete on the context and another complete on the unsubscribe future.

Reply:
Good point, I guess it will be enough just call `context.awaitCompletion(HTTP_REQUEST_TIMEOUT, TimeUnit.SECONDS)` instead of having CompletableFuture in these methods.